src/main/java/org/prebid/server/settings/service/JdbcPeriodicRefreshService.java

```diff
@@ -0,0 +1,199 @@
+ */
+public class JdbcPeriodicRefreshService {
+
+    private static final Logger logger = LoggerFactory.getLogger(JdbcPeriodicRefreshService.class);
+
+    private final CacheNotificationListener cacheNotificationListener;
+    private final Vertx vertx;
+    private final JdbcClient jdbcClient;
+    private final long refreshPeriod;
+
+    /**
+     * Example of initialize query:
+     * <pre>
+     * SELECT id, requestData, type
+     * FROM stored_requests;
+     * <pre>
+     * This query will be run once on startup to fetch _all_ known Stored Request data from the database.
+     */
+    private final String initQuery;
+
+    /**
+     * Example of initialize query:
+     * <pre>
+     * SELECT id, requestData, type
+     * FROM stored_requests
+     * WHERE last_updated > ?;
+     * <pre>
+     * The code will be run periodically to fetch updates from the database.
+     * Wildcard "?" would be used to pass last update date automatically.
+     */
+    private final String updateQuery;
+    private final TimeoutFactory timeoutFactory;
+    private final long timeout;
+    private Instant lastUpdate;
+
+    public JdbcPeriodicRefreshService(CacheNotificationListener cacheNotificationListener,
+                                      Vertx vertx, JdbcClient jdbcClient, long refreshPeriod, String initQuery,
+                                      String updateQuery, TimeoutFactory timeoutFactory, long timeout) {
+        this.cacheNotificationListener = Objects.requireNonNull(cacheNotificationListener);
+        this.vertx = Objects.requireNonNull(vertx);
+        this.jdbcClient = Objects.requireNonNull(jdbcClient);
+        this.refreshPeriod = refreshPeriod;
+        this.initQuery = Objects.requireNonNull(StringUtils.stripToNull(initQuery));
+        this.updateQuery = Objects.requireNonNull(StringUtils.stripToNull(updateQuery));
+        this.timeoutFactory = Objects.requireNonNull(timeoutFactory);
+        this.timeout = timeout;
+    }
+
+    public void initialize() {
+        getAll();
+        if (refreshPeriod > 0) {
+            vertx.setPeriodic(refreshPeriod, aLong -> refresh());
+        }
+    }
+
+    private void getAll() {
+        lastUpdate = Instant.now();
+
+        jdbcClient.executeQuery(initQuery, Collections.emptyList(),
+                JdbcPeriodicRefreshService::mapToStoredRequestResult, createTimeout())
+                .map(this::save)
+                .recover(JdbcPeriodicRefreshService::failResponse);
+    }
+
+    private Void save(StoredDataResult storedDataResult) {
+        cacheNotificationListener.save(storedDataResult.getStoredIdToRequest(), storedDataResult.getStoredIdToImp());
+        return null;
+    }
+
+    private static Future<Void> failResponse(Throwable exception) {
+        logger.warn("Error occurred while request to jdbc refresh service", exception);
+        return Future.failedFuture(exception);
+    }
+
+    private static StoredDataResult mapToStoredRequestResult(ResultSet resultSet) {
```

Comment:
We can fight with code duplication by creating class like `JdbcStoredDataResultMapper` with one method `map`.
This method will return `StoredDataResult` object. Pls use implementation from `org.prebid.server.settings.JdbcApplicationSettings#mapToStoredRequestResult`.

PS. initialize `JdbcStoredDataResultMapper` in constructor of `JdbcApplicationSettings` & `JdbcPeriodicRefreshService`.

PS2. Pls check the same solution for `HttpApplicationSettings` & `HttpPeriodicRefreshService`.

PS3. For sure we can create `XxxUtil` class but it is not so OOP :)